runtime/etc/debug/chpl_lldb_pretty_print.py

```diff
@@ -0,0 +1,543 @@
+        if index < 0 or index >= self.num_children():
+            return None
+        key = list(self.synthetic_children.keys())[index]
+        return self.valobj.CreateValueFromExpression(
+            key, self.synthetic_children[key].GetValue()
+        )
+
+
+# TODO: only handles DefaultRectangularDom for now
+domain_regex_c = re.compile(
+    r"^_domain_DefaultRectangularDom_([0-9]+)_([a-zA-Z0-9_]+)_(one|negOne|positive|negative|any)_chpl$"
+)
+domain_regex_llvm = re.compile(r"^$")  # TODO
+
+
+def DomainRecognizer(sbtype, internal_dict):
+    typename = sbtype.GetName()
+    match = domain_regex_c.match(typename) or domain_regex_llvm.match(typename)
+    return match is not None
+
+
+def DomainSummary(valobj, internal_dict):
+    typename = valobj.GetTypeName()
+    match = domain_regex_c.match(typename) or domain_regex_llvm.match(typename)
+    if not match:
+        print(
+            f"Domain_Summary: type name '{typename}' did not match expected pattern"
+        )
+        return None
+
+    rank = int(match.group(1))
+    idx_type = match.group(2)
+    stride_kind = match.group(3)
+
+    ranges = (
+        valobj.GetNonSyntheticValue()
+        .GetChildMemberWithName("_instance")
+        .GetNonSyntheticValue()
+        .GetChildMemberWithName("ranges")
+    )
+    return "{" + (ranges.GetSummary()) + "}"
+
+
+class DomainProvider:
+    def __init__(self, valobj, internal_dict):
+        self.valobj = valobj
+
+        self.synthetic_children = {}
+        typename = self.valobj.GetTypeName()
+        match = domain_regex_c.match(typename) or domain_regex_llvm.match(
+            typename
+        )
+        if not match:
+            print(
+                f"DomainProvider: type name '{typename}' did not match expected pattern"
+            )
+            return
+
+        ranges = (
+            self.valobj.GetNonSyntheticValue()
+            .GetChildMemberWithName("_instance")
+            .GetNonSyntheticValue()
+            .GetChildMemberWithName("ranges")
+        )
+        self.synthetic_children["dim"] = ranges
+
+    def has_children(self):
+        return self.num_children() > 0
+
+    def num_children(self):
+        return len(self.synthetic_children)
+
+    def get_child_index(self, name):
+        if name in self.synthetic_children:
+            return list(self.synthetic_children.keys()).index(name)
+        else:
+            return -1
+
+    def get_child_at_index(self, index):
+        if index < 0 or index >= self.num_children():
+            return None
+        key = list(self.synthetic_children.keys())[index]
+        child_obj = self.synthetic_children[key]
+        return self.valobj.CreateValueFromAddress(
+            key, child_obj.GetLoadAddress(), child_obj.GetType()
+        )
+
+
+#
+# homogenous tuples
+#
+homogeneous_tuple_regex_c = re.compile(
+    r"^_tuple_([0-9]+)_star_([a-zA-Z0-9_]+)_chpl$"
+)
+homogeneous_tuple_regex_llvm = re.compile(r"^$")  # TODO
```

Comment:
I'm a little unclear of what the value is of providing this, since it's not really implemented